### PR TITLE
Enable routing and accept cluster name in install script

### PR DESCRIPTION
**Pull request by DTMad**

### CODE REVIEW

File: deploy/install.sh [hunk 1]
```diff
@@ -50,6 +56,21 @@ if [ -z "$PAAS_TOKEN" ]; then
   exit 1
 fi
 
+if [ -z "$CLUSTER_NAME" ]; then
+  echo "Error: cluster name not set!"
+  exit 1
+fi
+
+if ! echo "$CLUSTER_NAME" | grep -q "$CLUSTER_NAME_REGEX"; then
```
Comment (lrgar) [inline deploy/install.sh:64]: `grep` doesn't match regex by default, you want to use `-E` in Linux. No idea about Mac.

Bash also supports regexes, but you'd need to update the shebang.

File: deploy/install.sh [hunk 2]
```diff
@@ -146,14 +171,28 @@ EOF
...
 
+checkForExistingCluster() {
+  response="$(curl -sS -X GET "${API_URL}/config/v1/kubernetes/credentials" \
+    -H "accept: application/json; charset=utf-8" \
+    -H "Authorization: Api-Token ${API_TOKEN}" \
+    -H "Content-Type: application/json; charset=utf-8")"
+
+  if echo "$response" | grep -Fqe "\"name\":\"${CLUSTER_NAME}\""; then
+    echo "Error: Cluster already exists!"
+    exit 1
```
Comment (lrgar) [inline deploy/install.sh:189]: Why exit? Basically forces the customer to delete entry from Dynatrace to run the command twice, even if there is no change for the connection at all.

File: deploy/install.sh [hunk 3]
```diff
@@ -146,14 +171,28 @@ EOF
...
   fi
 }
 
+checkForExistingCluster() {
+  response="$(curl -sS -X GET "${API_URL}/config/v1/kubernetes/credentials" \
+    -H "accept: application/json; charset=utf-8" \
+    -H "Authorization: Api-Token ${API_TOKEN}" \
+    -H "Content-Type: application/json; charset=utf-8")"
+
+  if echo "$response" | grep -Fqe "\"name\":\"${CLUSTER_NAME}\""; then
```
Comment (lrgar) [inline deploy/install.sh:187]: I'm only accepting a customer argument to be used as a regex on this command because it's being validated before.

**Review by lrgar — CHANGES_REQUESTED** [review]

File: deploy/install.sh (hunk 1 above)
Comment (DTMad) [reply deploy/install.sh:64]: Added -E

File: deploy/install.sh (hunk 2 above)
Comment (DTMad) [reply deploy/install.sh:189]: This is on purpose. We will not overwrite customer configuration. In case he provides the same name he needs to adapt

**Review by lrgar — APPROVED** [review]

**Pull request merged**